src/v0/destinations/zendesk/transform.js

```diff
@@ -530,25 +565,41 @@ async function processTrack(message, destinationConfig, headers) {
   }
   let zendeskUserID;
 
-  let url = `${endPoint}users/search.json?query=${userEmail}`;
+  const url = `${baseEndpoint}users/search.json?query=${userEmail}`;
   const config = { headers };
-  const userResponse = await axios.get(url, config);
-  if (!get(userResponse, "data.users.0.id") || userResponse.data.count === 0) {
-    const { zendeskUserId, email } = await createUser(
-      message,
-      headers,
-      destinationConfig
-    );
-    if (!zendeskUserId) {
-      throw new NetworkInstrumentationError("User not found");
-    }
-    if (!email) {
-      throw new NetworkInstrumentationError("User email not found", 400);
+  try {
+    const userResponse = await axios.get(url, config);
+    if (
+      !get(userResponse, "data.users.0.id") ||
+      userResponse.data.count === 0
+    ) {
+      const { zendeskUserId, email } = await createUser(
+        message,
+        headers,
+        destinationConfig,
+        baseEndpoint
+      );
+      if (!zendeskUserId) {
+        throw new NetworkInstrumentationError("User not found");
+      }
+      if (!email) {
+        throw new NetworkInstrumentationError("User email not found", 400);
+      }
+      zendeskUserID = zendeskUserId;
+      userEmail = email;
     }
-    zendeskUserID = zendeskUserId;
-    userEmail = email;
+    zendeskUserID = zendeskUserID || userResponse.data.users[0].id;
+  } catch (error) {
+    throw new NetworkError(
```

Comment:
Whenever a network error is being sent out we need to send the raw destination response as part of the argument
```
throw new NetworkError(
`Failed to fetch user with email: ${userEmail} due to ${JSON.stringify(
        error.response ? error.response.data : error
      )}`,
  error.status,
  {
    [tags.TAG_NAMES.ERROR_TYPE]: getDynamicErrorType(error.status)
  },
  error?.response?.data || error
);
```